tests/console/rails.pm

```diff
@@ -0,0 +1,28 @@
+# SUSE's openQA tests
+#
+# Copyright © 2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+use strict;
+use base "consoletest";
+use testapi;
+
+sub run() {
+    select_console 'root-console';
+    my $cmd = <<'EOF';
+zypper -n in -C "rubygem(rails)"
+rails new -B mycoolapp
+cd mycoolapp
+(rails server &)
+for i in {1..100} ; do sleep 0.1; curl -s http://localhost:3000 | grep "Welcome" && break ; done
```

Comment:
I see three realistic cases:

1. `curl` never succeeds in reaching port 3000 --> all 100 `curl` calls will fail, for loop returns 1 (exit code of 100th `curl`)
2. `curl` eventually succeeds but all `grep` fail because there is no Welcome --> for loop returns 1 (exit code of last `grep`)
3. loop continues until one `curl` plus one `grep` succceed, `break` is called, for loop returns 0 (exit code of last `grep`, `break` does not alter exit code)

or did I get you wrong? You could check with `for i in {1..100} ; do false && break ; done` and `for i in {1..100} ; do true && break ; done`